src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -386,6 +410,20 @@ void mp::QemuVirtualMachine::on_restart()
 
 void mp::QemuVirtualMachine::ensure_vm_is_running()
 {
+    if (is_starting_from_suspend)
+    {
+        // Due to https://github.com/canonical/multipass/issues/2374, the DHCP address is removed from
+        // the dnsmasq leases file, so if the daemon restarts while an instance is suspended and then
+        // starts the instance, the daemon won't be able to reach the instance since the instance
+        // won't refresh its IP address.  The following will force the instance to refresh by resetting
+        // the network every 30 seconds until the start timoeut is reached.
+        if (std::chrono::steady_clock::now() > current_resume_start_time + 30s)
```

Comment:
Should we do this sooner after resume? Normally resume takes mere seconds, with this it will only work after 30s. Maybe we could do first at 5s, then at 30s intervals?

Reply:
It's just an arbitrary time, so I guess 5s would be enough.  I think how fast an instance resumes really depends on the load of the system, but sure I can do 5s and then 30s thereafter.

Reply:
Of course, this will mean, though, that best case scenario is reduced to 5s, worst case is only delayed by 16.3(3)% :)

Reply:
Ok, done.